app/services/authenticator.ts

```diff
@@ -72,20 +72,20 @@ export default class AuthenticatorService extends Service {
     await this.syncCurrentUser();
   }
 
-  initiateLogin(redirectPath: string | null) {
-    const isFullUrl = redirectPath?.startsWith(`${config.x.backendUrl}`);
-
-    if (isFullUrl) {
-      window.location.href = `${config.x.backendUrl}/login?next=` + redirectPath;
-    } else if (redirectPath) {
-      window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(`${window.origin}${redirectPath}`);
-    } else if (this.router.currentURL) {
+  initiateLogin() {
+    if (this.router.currentURL) {
       window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(`${window.origin}${this.router.currentURL}`);
     } else {
       window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(window.origin);
     }
   }
 
+  initiateLoginAndRedirectTo(redirectPathOrUrl: string) {
+    const fullRedirectUrl = redirectPathOrUrl.startsWith('/') ? `${window.origin}${redirectPathOrUrl}` : redirectPathOrUrl;
```

Comment:
```suggestion
    const redirectUrl = redirectPathOrUrl.startsWith('/') ? `${window.origin}${redirectPathOrUrl}` : redirectPathOrUrl;
```

("url" vs "path" is already clear, so just `redirectUrl` should do here? Another option is `finalRedirectUrl` to signify that there's an intermediary redirect followed by another. "Full" doesn't add much meaning)